aws/modules/vpc/main.tf

```
@@ -0,0 +1,71 @@
+data "aws_availability_zones" "available_zones" {
+  state = "available"
+}
+
+resource "aws_vpc" "business_service" {
+  cidr_block           = var.cidr_block
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+  tags                 = var.tags
+}
+
+resource "aws_subnet" "public" {
+  count                   = length(var.public_subnets) > 0 ? length(var.public_subnets) : 0
+  cidr_block              = element(var.public_subnets, count.index)
+  availability_zone       = data.aws_availability_zones.available_zones.names[count.index]
+  vpc_id                  = aws_vpc.business_service.id
+  map_public_ip_on_launch = true
+  tags                    = var.tags
+}
+
+resource "aws_subnet" "private" {
+  count             = length(var.private_subnets) > 0 ? length(var.private_subnets) : 0
+  cidr_block        = element(var.private_subnets, count.index)
+  availability_zone = data.aws_availability_zones.available_zones.names[count.index]
+  vpc_id            = aws_vpc.business_service.id
+  tags              = var.tags
+}
+
+resource "aws_internet_gateway" "gateway" {
+  vpc_id = aws_vpc.business_service.id
+  tags   = var.tags
+}
+
+resource "aws_route" "internet_access" {
+  count                  = length(var.public_subnets) > 0 ? 1 : 0
+  route_table_id         = aws_vpc.business_service.main_route_table_id
+  destination_cidr_block = local.all_ips
+  gateway_id             = aws_internet_gateway.gateway.id
+}
+
+resource "aws_eip" "nat_gateway" {
+  count      = length(var.private_subnets) > 0 ? length(var.private_subnets) : 0
+  vpc        = true
+  depends_on = [aws_internet_gateway.gateway]
+  tags       = var.tags
+}
+
+resource "aws_nat_gateway" "this" {
+  count         = length(var.private_subnets) > 0 ? length(var.private_subnets) : 0
+  subnet_id     = element(aws_subnet.public.*.id, count.index)
+  allocation_id = element(aws_eip.nat_gateway.*.id, count.index)
+  tags          = var.tags
+}
+
+resource "aws_route_table" "private" {
+  count  = length(var.private_subnets) > 0 ? length(var.private_subnets) : 0
```

to make it a bit easier to read we could define it first in a local and use the local here 

```hcl
locals {
  priv_subnet_count = length(var.private_subnets)
}
```

```suggestion
  count  = priv_subnet_count > 0 ? priv_subnet_count : 0
```